grpc/interceptors.go

```diff
@@ -22,6 +23,28 @@ func (si *serverInterceptor) intercept(ctx context.Context, req interface{}, inf
 	if info == nil {
 		return nil, berrors.InternalServerError("passed nil *grpc.UnaryServerInfo")
 	}
+
+	// Shave 100 milliseconds off the deadline to ensure that the RPC server times
+	// out any sub-calls it makes (like DNS lookups, or onwards RPCs), it has a
+	// chance to report that timeout to the client. This allows for more specific
+	// errors, e.g "the VA timed out looking up CAA for example.com" (when called
+	// from RA.NewCertificate, which was called from WFE.NewCertificate), as
+	// opposed to "RA.NewCertificate timed out" (causing a 500).
+	// Once we've shaved the deadline, we ensure we have we have at least another
+	// 100ms left to do work; otherwise we abort early.
+	deadline, ok := ctx.Deadline()
+	// Should never happen: there was no deadline.
```

Comment:
Can you add a warning log or equivalent here? If it does happen it would be best if we had a way to know which RPC it was, otherwise this could be pretty mysterious to track down.

Reply:
The interceptors don't currently have a logger field, so this would be a bunch of work. I feel quite confident that this case won't happen, because our client interceptor always assigns a timeout.